clients/src/main/java/org/apache/kafka/common/protocol/ApiVersion.java

```diff
@@ -53,4 +53,18 @@ public String toString() {
             ", maxVersion= " + maxVersion +
             ")";
     }
+
+    public static ApiVersion versionsInCommon(ApiKeys apiKey, ApiVersion supportedVersions,
```

Comment:
nit: we could simplify this a little bit. I'd suggest a non-static method. Maybe something like this:
```java
public ApiVersion intersect(ApiVersion other) {
  // verify this.apiKey matches other.apiKey
  // return intersection as implemented below
}
```

Reply:
I looked up the usage of this API, it has the external effect to have reference `min` and `max` versions. At least the current format is the most straightforward refactoring without the need to create another `ApiVersion` instance.